dev/com.ibm.ws.jaxrs.2.0.common/src/com/ibm/ws/jaxrs20/security/LibertySimpleAuthorizingInterceptor.java

```diff
@@ -29,17 +31,24 @@
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
 
+@Priority(Priorities.AUTHORIZATION)
 public class LibertySimpleAuthorizingInterceptor extends
                 AbstractAuthorizingInInterceptor {
     private static final TraceComponent tc = Tr
                     .register(LibertySimpleAuthorizingInterceptor.class);
 
     @Override
     public void handleMessage(Message message) throws Fault {
-        SecurityContext sc = message.get(SecurityContext.class);
-        if (sc != null) {
+        SecurityContext cxfSecurityContext = message.get(SecurityContext.class);
+        javax.ws.rs.core.SecurityContext jaxrsSecurityContext = message.get(javax.ws.rs.core.SecurityContext.class);
```

Comment:
Instead of duplicating the existing methods (using CXF's `SecurityContext`), maybe we could create a subclass of CXF's `SecurityContext` that delegates appropriately - something like:
```
class SecurityContextProxy implements SecurityContext {
    private final javax.ws.rs.core.SecurityContext jaxrsSecCtx;
    private final SecurityContext cxfSecCtx;
    SecurityContextProxy(javax.ws.rs.core.SecurityContext jaxrsSecCtx, SecurityContext cxfSecCtx) {
        this.jaxrsSecCtx = jaxrsSecCtx;
        this.cxfSecCtx = cxfSecCtx;
        if (jaxrsSecCtx == null && cxfSecCtx == null) throw new IllegalArgumentException();
    }

    @Override
    public Principal getUserPrincipal() {
        return jaxrsSecCtx != null ? jaxrsSecCtx.getUserPrincipal() : cxfSecCtx.getUserPrincipal();
    }

    @Override
    public boolean isUserInRole(String role) {
        return jaxrsSecCtx != null ? jaxrsSecCtx.isUserInRole(role) : cxfSecCtx.isUserInRole(role);
    }
}
```

This would allow you to keep the existing `parse*` methods and just pass an instance of this new proxy class to it - without duplicating for both SecurityContext types.